### PR TITLE
OP-1820: Building upgrade package release candidates and staging on S3.

### PR DESCRIPTION
**Pull request by sacherjj**
This builds `bin.tar.gz` and `config.tar.gz`.

`config.tar.gz` includes `chainspec.toml`, `accounts.csv`, and `config-example.toml` from `/resources/production`.

`bin.tar.gz` includes `casper-node` built for Ubuntu 18.04 and `README.md` with repo link and github commit hash used to build, to allow other platforms to build casper-node from same code.

Upload script will place these artifacts at http://genesis.casperlabs.io/drone/[git-hash]/[protocol_version].  Where protocol_version is from `resources/production/chainspec.toml` with underscores replacing periods.  

These can be used in upgrade tests and one can be selected for network upgrade and moved into the correct location in the `genesis.casperlabs.io` S3 bucket.

### CODE REVIEW

**Comment by thamps-casp** [comment]
this change forward for dev/test/prod:
- install 'casper-node-launcher' debian package
- 'casper-node-launcher' downloads latest config and bin packages from drone/[drone_build_number]
- Semver set for casper-node-launcher
- No 'casper-node' package builds and wasm artifacts published to 'casperlabs-cicd-artifacts'

File: ci/build_update_package.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+build_casper_node() {
+    echo "Building casper-node"
+    cd "$NODE_BUILD_DIR"
+    cargo build --release
+}
+
+make_bin_readme() {
+    mkdir -p "$BIN_DIR"
+    echo "Generating bin README.md"
```
Comment (sacherjj) [inline ci/build_update_package.sh:38]:
```suggestion
    echo "Generating bin README.md"
    mkdir -p "$BIN_DIR"
```

**Comment by sacherjj** [comment]
Updated to move S3 location to `genesis.casperlabs.io/drone/[git-hash]/[protocol_version]/`  This will allow use of `sudo -u casper /etc/casper/pull_casper-node_version.sh [protocol_version] drone/[git_hash]`

File: .drone.yml [hunk 2]
```diff
@@ -224,6 +245,7 @@ steps:
   # bors should make sure, that it has passed on staging or trying branches
 - name: publish-test-bintray
   image: casperlabs/node-build-u1804
+  # Keeping casper-node from int test publish to allow networks stood up without casper-node-launcher for now.
```
Comment (TomVasile) [inline .drone.yml:248]: nit delete this

File: .drone.yml [hunk 3]
```diff
@@ -325,15 +364,20 @@ steps:
 
 - name: publish-prod-bintray
   image: casperlabs/node-build-u1804
+  # Removed casper-node from tag publish as we not longer ship this way.
   commands:
-  - "./upload.sh --repo-name debian --package-name casper-node --package-tag true"
```
Comment (TomVasile) [inline .drone.yml:329]: i feel this should be added back unless this uploaded in a different automation step

Comment (sacherjj) [reply .drone.yml:329]: This was published to participate in our networks, which will be delivered via bin.tar.gz now.  Installing the deb will not be compatible.

File: ci/upgrade_package_s3_storage.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+if [[ "$ACTION" == "del" ]]; then
+  LOCAL=$2
+
+  if [ -z "$LOCAL" ]; then
+    echo "Source not provided"
+    exit 1
+  fi
+
+  if [ -z "$LOCAL" ]; then
```
Comment (thamps-casp) [inline ci/upgrade_package_s3_storage.sh:46]: duplicate param check

File: ci/upgrade_package_s3_storage.sh [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+export AWS_ACCESS_KEY_ID
+AWS_SECRET_ACCESS_KEY=$(echo "$CREDENTIALS" | jq -r .data.cicd_agent_to_s3.aws_secret_key)
+export AWS_SECRET_ACCESS_KEY
+
+CL_S3_BUCKET="genesis.casperlabs.io"
+CL_S3_LOCATION="drone/$GIT_HASH"
+
+case "$ACTION" in
+"put")
+  echo "sync ${LOCAL} s3://${CL_S3_BUCKET}/${CL_S3_LOCATION}/${PROTOCOL_VERSION}"
```
Comment (thamps-casp) [inline ci/upgrade_package_s3_storage.sh:68]: LOCAL not set for these actions. 
for s3 sync target url needs to end with a '/' to upload under path i beleive.

**Review by rob-casper — APPROVED** [review]

**Review by thamps-casp — APPROVED** [review]

**Pull request merged**